src/components/hv-view/index.js

```diff
@@ -122,17 +131,25 @@ export default class HvView extends PureComponent<HvComponentProps> {
       }
     }
 
-    // $FlowFixMe
-    const component = React.createElement(
-      c,
-      props,
-      ...Render.renderChildren(
-        this.props.element,
-        this.props.stylesheets,
-        this.props.onUpdate,
-        viewOptions,
-      ),
+    const children = Render.renderChildren(
+      this.props.element,
+      this.props.stylesheets,
+      this.props.onUpdate,
+      viewOptions,
     );
+
+    if (scrollable && scrollDirection !== 'horizontal') {
+      // add sticky indicies
+      // $FlowFixMe
```

Comment:
`child` might be a `string` or `any` type, which is why you're getting a flow error - you need to check for the type and presence of `props` and `element`.
Looking back at the evolution of the change, it's now possible to simplify by iterating over the list of children and generating the indices in a single pass: the array reduce can be done over the `children` list directly. Note also the typo in "indices"

```js
const stickyIndices = children.reduce(
  (acc, element, index) =>
    typeof element !== 'string' &&
    element.props?.element?.getAttribute('sticky') === 'true'
      ? [...acc, index]
      : acc,
  [],
);

Reply:
Gotcha. Curious, in which cases would a `child` be a `string`?

Reply:
If you drill down into `Render.renderChildren`, which internally calls `renderElement`, you can see the code path that could lead to rendering a string. Specifically, this would be these `return` statement: [1](https://github.com/Instawork/hyperview/blob/master/src/services/render/index.js#L108), [2](https://github.com/Instawork/hyperview/blob/master/src/services/render/index.js#L118)